SA3/qa_web/templates/qa_web/answerspage.html

```diff
@@ -49,9 +49,9 @@
          <h1 class="masthead-heading mb-0"><span id="title_question"><b>{{ currentQuestion.title }}</b></span> &nbsp; <span class="label label-primary">{{ currentQuestion.visits }} visits</span></h1>
           <h5 class="masthead-subheading mb-0"><p><span class="marked" id="content_question">{{ currentQuestion.content }}</span></p></h5>
         <h6>Question by {{ currentQuestion.owner }} on {{ currentQuestion.creation_date }}</h6>
-        <a id='upvote_{{ currentQuestion.id }}_question'><button class="btn btn-primary btn-xl rounded-pill mt-5">Upvote</button></a>
+        <a id='upvote_{{ currentQuestion.id }}_question' class="{% if pos_v_q %}voted{% endif %}"><button class="btn btn-primary btn-xl rounded-pill mt-5">Upvote</button></a>
```

Comment:
The logic here is flawed. This results in the button staying highlighted if one cancels their vote rather than switching from one to the other. Replace your logic to check that the question id is pos_v_q. Be sure to do the same for the downvote button right below this line of code.

Reply:
I'll make the change for that and rebase this branch for merge conflicts issues.

Reply:
The statement is fine: here `pos_v_q` is a boolean and not a list (as seen on line 186 of `views.py`) therefore the condition is `if pos_v_q` and not `if question.id in pos_v_q`.
The reason why `pos_v_q` is a boolean and not a list is because there is only 1 question per thread while there can be multiple answers and comments.

Reply:
This still leaves the problem of the button remaining highlighted when the user votes and cancels their vote without reloading the page though.

Reply:
With further observation, I concluded that the logic for this code is sound but I discovered the CSS will change the button color and add a border on the button when clicked even without this implementation at all.

Reply:
Yes, the color choice for the `.voted` class makes it a confusing considering what you mentionned. I will change the color for better distinction between a clicked button and a vote.

Reply:
Actually, the btn-primary method makes it show a certain color if you click it no matter what. Only when you click outside the button does the actual color appear. I tested this by changing the color to something obvious (lime-green)

The best way to go about this is to remove this method. This may remove the pink buttons that are prevalent where the question is posted but it may be for the best since btn-primary is screwing up the coloring for vote indication entirely.